File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> solution(string my_str, int n) {
    vector<string> answer;
    int count = 0;
    string temp = "";
    for (char c : my_str) {
        count++;
        temp += c;
        if (count == n) {
            answer.push_back(temp);
            count = 0;
            temp = "";
        }
    }
    if (!temp.empty()) {
        answer.push_back(temp);
    }
    return answer;
}

File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.py
def solution(my_str, n):
    answer = []
    temp = ""
    cnt = 0
    for c in my_str:
        temp += c
        cnt += 1
        if cnt == n:
            answer.append(temp)
            cnt = 0
            temp = ""    
    if temp:
        answer.append(temp)
    return answer